python/lsst/pipe/supertask/parser.py

```diff
@@ -0,0 +1,278 @@
+            description = textwrap.dedent("""\
+                Display various information about given task. By default all information is
+                displayed, use options to select only subset of the information.
+
+                Note: Specify task name with --help to also see task-specific options""")
+        else:
+            description = textwrap.dedent("""\
+                Execute specified task.
+
+                Note: Specify task name with --help to also see task-specific options""")
+
+        subparser = subparsers.add_parser(subcommand,
+                                          add_help=False,
+                                          usage='%(prog)s taskname [options]',
+                                          description=description,
+                                          epilog=_epilog,
+                                          formatter_class=RawDescriptionHelpFormatter)
+        subparser.set_defaults(config_overrides=[], subparser=subparser)
+        subparser.add_argument("taskname", nargs='?', default=None,
+                               help="Name of the task to execute. This can be a simple name without dots "
+                               "which will be found in one of the modules located in the known "
+                               "packages or packages specified in --package option. If name contains dots "
+                               "it is assumed to be a fully qualified name of a class found in $PYTHONPATH")
+        subparser.add_argument('-h', "--help", action="store_true", dest='do_help', default=False,
+                               help="Show this help message and exit")
+        subparser.add_argument("-c", "--config", dest="config_overrides", nargs="+",
+                               action=_AppendFlattenAction, type=_config_override,
+                               help="Configuration override(s), e.g. -c foo=newfoo bar.baz=3",
+                               metavar="NAME=VALUE")
+        subparser.add_argument("-C", "--configfile", dest="config_overrides", nargs="+",
+                               action=_AppendFlattenAction, type=_config_file,
+                               metavar="PATH", help="Configuration override file(s)")
+        subparser.add_argument("--show", metavar='ITEM|ITEM=VALUE', nargs="+", default=[],
+                               help="Dump various info to standard output. Possible items are: "
+                               "`config', `config=<PATTERN>' or `config=<PATTERN>:NOIGNORECASE' "
+                               "to dump configuration possibly matching given pattern; "
+                               "`history=<FIELD>' to dump configuration history for a field, "
+                               "field name is specified as Task.SubTask.Field; "
+                               "`data' to show information about data in butler; "
+                               "`tasks' to show task composition.")
+
+    return parser
```

Comment:
There is a lot of copy and paste in this code plus my own additions so it's not only "my" style 😄 I'll improve it.